### ISSUE TITLE
When using org.jooq.Meta, UniqueKey.getReferences throws NPE in Firebird

### ISSUE THREAD

**Issue by perceptron8** [jOOQ/jOOQ#7377]
### Expected behavior and actual behavior:
`UniqueKey.getReferences()` should return list of foreign keys, throws `NullPointerException` instead.

### Steps to reproduce the problem:
1. git clone https://github.com/perceptron8/jooq-7377.git
2. cd jooq-7377
3. gbak -r o -v -user SYSDBA -password masterkey src/test/resources/test.fbk src/test/resources/test.fdb
4. mvn test

### Versions:

- jOOQ: 3.10.6
- Java: 1.8.0_161
- Database: Firebird 3.0.2
- OS: Linux
- JDBC Driver: Jaybird 3.0.3

### Findings:

1. As Firebird doesn't support schemas, default (empty one) is returned:
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L295

2. It is indexed here using it's name (`""`):
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L866

3. `DatabaseMetaData.getExportedKeys` returns `null` in FKTABLE_SCHEM:
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L833
so at:
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L870
`entry.getKey().get(1)` (equal to FKTABLE_SCHEM) equals `null`. 

4. Because `schemas.get(null)` returns `null`, following dereference ends up with `NullPointerException`:
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L872

In short: `null` argument given to `schemas.get(...)` comes from FKTABLE_SCHEM returned by `DatabaseMetaData.getExportedKeys`. This seems to be perfeclty valid according to javadoc:
https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html#getExportedKeys-java.lang.String-java.lang.String-java.lang.String-

Replacing
`result.add(new MetaSchema("", MetaCatalog.this));`
with
`result.add(new MetaSchema(null, MetaCatalog.this));`
at
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L295
resolves this issue, but it's up to you to check if it breaks something else at the same time (I hope it's fair).

---

Edit: jooq-7373 renamed to jooq-7377

**Comment by perceptron8** [comment]
BTW: Unrelated, but possible source of another problems:
https://github.com/jOOQ/jOOQ/blob/version-3.10.6/jOOQ/src/main/java/org/jooq/impl/MetaImpl.java#L184

**Comment by lukaseder** [comment]
Fixed (finally!) in jOOQ 3.13.0. Backport scheduled for 3.12.2 (#9422)

**Comment by perceptron8** [comment]
@lukaseder Thanks!
